src/addons/plugin/plugin.js

```diff
@@ -39,16 +39,21 @@ class Plugin {
     this.ipcSocket.bind();
     this.exec = '';
     this.execPath = '.';
-    this.process = null;
+
+    // Make this a nested object such that if the Plugin object is reused,
+    // i.e. the plugin is disabled and quickly re-enabled, the gateway process
+    // can maintain a proper reference to the process object.
+    this.process = {p: null};
```

Comment:
This seems to be causing a test failure

Reply:
The plugin-test.js is looking for plugin.process.on

Reply:
Fixed.